src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs

```diff
@@ -32,7 +32,7 @@ public async Task<UpsertResult> UpsertDocument(IUpsertRequest upsertRequest)
         try
         {
             await using var connection = await _dataSource.OpenConnectionAsync();
-            await using var transaction = await connection.BeginTransactionAsync(IsolationLevel.Serializable);
+            await using var transaction = await connection.BeginTransactionAsync(IsolationLevel.RepeatableRead);
```

Comment:
As mentioned, serializable seems overly prone to false-positives, especially on smaller table sizes. Repeatable read means a transaction gets a snapshot at the start of the transaction that never changes (unlike read committed where concurrent transaction commits change data mid-transaction). Like serializable but more narrowly focused, two concurrent transactions touching the same thing results in a failure for one, requiring a retry.

Reply:
I like the sound of this better. This is still more strict than the ODS/API, so I'm leaving the spike ticket open as a reminder to check in with some stakeholders to see if they have strong feelings about this. If nothing else, I'm leaning toward making this configurable, with `RepeatableRead` as a sensible default.